release_notes.md

```diff
@@ -1,3 +1,71 @@
+**********************************
+# Release Notes for OpenMDAO 3.1.0
+
+May 01, 2020
+
+OpenMDAO 3.1.0 is a periodic release of OpenMDAO.
+This release features more uniformity in recording options of Problem, Driver, System, and Solver.
+It adds more capability to the Problem recorder for when users only wish to record the final state of the model rather than iteration-by-iteration.
+There have been numerous updates relating to the N2 viewer and distributed components.
+
+## Backwards Incompatible API Changes:
+
+- <POEM001> Updates the definition of light year (ly) to the IAU definition and adds units for astronomical units (au) and parsec (pc).  #1204
+- <POEM014> Build-in XDSM viewer removed from OpenMDAO and is now supported as an external plugin. #1240
+- assert_rel_error deprecated replaced with assert_near_equal. #1192
+- <POEM017> Adds ability to specify units on design variables, constraints, and objectives. #1265
+- <POEM013> convert_units and unit_conversion added to openmdao.api. #1267
+- <POEM019> Fixed various issues with directional derivatives. #1314
+- Renamed Problem.record_state to Problem.record. #1355
```

Comment:
`record_state` wasn't in 3.0.. it had a short-lived existence from Mar 23 (#1269) to April 23 (#1355)   
The method was `record_iteration` in 3.0 and that was deprecated by #1269